components/ActionSheet/src/private/MDCActionSheetHeaderView.m

```diff
@@ -0,0 +1,258 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCActionSheetHeaderView.h"
+
+#import "MaterialTypography.h"
+
+static const CGFloat kTitleLabelAlpha = 0.87f;
+static const CGFloat kMessageLabelAlpha = 0.6f;
+static const CGFloat kMessageOnlyPadding = 23.f;
+static const CGFloat kLeadingPadding = 16.f;
+static const CGFloat kStandardPadding = 16.f;
+static const CGFloat kTrailingPadding = 16.f;
+static const CGFloat kTitleOnlyPadding = 18.f;
+static const CGFloat kMiddlePadding = 8.f;
+
+@interface MDCActionSheetHeaderView ()
+
+@property(nonatomic, nonnull, strong) UIScrollView *scrollView;
+
+@end
+
+@implementation MDCActionSheetHeaderView {
+  UILabel *titleLabel;
+  UILabel *messageLabel;
+}
+
+@synthesize mdc_adjustsFontForContentSizeCategory;
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCActionSheetHeaderViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithTitle:(NSString *)title {
+  return [[MDCActionSheetHeaderView alloc] initWithTitle:title message:nil];
+}
+
+- (nonnull instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super init];
+  if (self) {
+    self.title = title;
+    self.message = message;
+    [self commonMDCActionSheetHeaderViewInit];
+  }
+  return self;
+}
+
+- (void)commonMDCActionSheetHeaderViewInit {
+  self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
+  titleLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+  messageLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+  [self addSubview:self.scrollView];
+  [self.scrollView addSubview:titleLabel];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleSubheadline];
+  } else {
+    titleLabel.font = [MDCTypography subheadFont];
+  }
+  titleLabel.numberOfLines = 0;
+  titleLabel.lineBreakMode = NSLineBreakByTruncatingMiddle;
+
+  [self.scrollView addSubview:messageLabel];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    messageLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleBody1];
+  } else {
+    messageLabel.font = [MDCTypography body1Font];
+  }
+  messageLabel.numberOfLines = 2;
+  messageLabel.lineBreakMode = NSLineBreakByWordWrapping;
+  messageLabel.alpha = kMessageLabelAlpha;
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  if (self.message == nil || [self.message isEqualToString:@""]) {
+    titleLabel.alpha = kMessageLabelAlpha;
+  } else {
+    titleLabel.alpha = kTitleLabelAlpha;
+  }
+  CGSize boundsSize = CGRectInfinite.size;
+  boundsSize.width = [self accomodateSafeAreaInWidth:CGRectGetWidth(self.bounds)];
```

Comment:
If we only care about the width, consider simplifying CGRectGetWidth(CGRectInfinite) to CGFLOAT_MAX

Reply:
We need the height as well to do sizeThatFits below